prisma/schema.prisma

```diff
@@ -0,0 +1,144 @@
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+// Users that are either authors, contributors or participants in slack threads, issues or pull requests
+model User {
+  id             Int     @id @default(autoincrement())
+  githubUsername String?
+  email          String  @unique
+  slackId        String? @unique
+
+  messagesAsked      SlackMessage[]
+  issuesOpened       GithubItem[]
+  itemsResolved      ActionItem[]
+  participatingItems Participant[]
```

Comment:
There's no consistent naming convention being followed. `Participant` becomes `participatingItems` but `ActionItem` relationship is called `itemsResolved`?

Another better way to name associations could be:

```
  slackMessages SlackMessage[]
  githubItems   GithubItem[]
  actionItems   ActionItem[]
  participants  Participant[]
```

Reply:
hmm, i focused on readability rather than convention here. so `user.participatingItems` is the actual list of items where the user is a participant. same for other fields.

Reply:
`messagesAsked` doesn't sound grammatically accurate. It can be just `messages` or `messagesSent` (i feel that verb is redundant).

We have `participants Participant[]` in `ActionItem` model but `participatingItems Participant[]` here? Seems inconsistent.

I saw that `ActionItem` is related to `User` via `resolvedBy` relationship. So it would make more sense to call the association here `resolvedItems`. You see the symmetry resolvedBy -> resolvedItems?

Reply:
https://martinfowler.com/bliki/TwoHardThings.html